### ISSUE TITLE
Document building tensorflow-gpu from source?

### ISSUE THREAD

**Issue by morrisonlevi** [tensorflow/tensorflow#11193]
I have built Tensorflow 1.2 from source with GPU support. However the package name is "tensorflow" and not "tensorflow-gpu" as is common. We also have machines that don't have GPUs in them so I'd like to have them both. However I could not find any documentation on how to do this when building from source.

**Comment by printdhruv** [comment]
Hi,
     It's already here -> [https://www.tensorflow.org/install/install_sources#ConfigureInstallation](https://www.tensorflow.org/install/install_sources#ConfigureInstallation). You need to have `bazel, GPU prerequisites` in order to execute `./configure`. where you will be asked following questions. 
```
Do you wish to build TensorFlow with CUDA support? [y/N] Y     
CUDA support will be enabled for TensorFlow
Please specify which gcc should be used by nvcc as the host compiler. [Default is /usr/bin/gcc]:
Please specify the Cuda SDK version you want to use, e.g. 7.0. [Leave empty to use system default]: 8.0
Please specify the location where CUDA 8.0 toolkit is installed. Refer to README.md for more details. [Default is /usr/local/cuda]:
Please specify the cuDNN version you want to use. [Leave empty to use system default]: 5
Please specify the location where cuDNN 5 library is installed. Refer to README.md for more details. [Default is /usr/local/cuda]:
Please specify a list of comma-separated Cuda compute capabilities you want to build with.
You can find the compute capability of your device at: https://developer.nvidia.com/cuda-gpus.
```

This third party link might help too, [https://alliseesolutions.wordpress.com/2016/09/08/install-gpu-tensorflow-from-sources-w-ubuntu-16-04-and-cuda-8-0/](https://alliseesolutions.wordpress.com/2016/09/08/install-gpu-tensorflow-from-sources-w-ubuntu-16-04-and-cuda-8-0/)

**Comment by morrisonlevi** [comment]
No, that's *not* what I'm after. I've went through that and confirmed I have a working tensorflow + GPU package. The issue is that the python module is named `tensorflow` and not `tensorflow-gpu`.

**Comment by printdhruv** [comment]
I misunderstood your question by reading heading only. I think if you try to install a common tensor-flow on non GPU machine . It won't have `CUDA , cuDNN`. So, it is obvious that will use CPU resources.
`./configure` ask you CUDA support question which is sufficient for non GPU machines for simplicity.
I believe for making development and distribution compatible and easy with each other they have singular source and for other use cases I guess. 

Here are some points in order to modify source : -

1. Try to experiment by removing `third_party/gpus` folder.

2. The easiest way to achieve this by comparing tensor flow-cpu vs tensor flow distribution source side by side and make your own version of tensoflow cpu by making changes in source. It's experimental and not sure will work or not.

**Comment by andydavis1** [comment]
@jart any ideas on this one?

**Comment by gunan** [comment]
At the moment, this is not possible. You will need to conditionally install CPU or GPU packages on each machine based on existence of GPU on them.

CC @martinwicke

**Comment by Mistobaan** [comment]
I think you just need to  invoke `bazel build -- //tensorflow/tools/pip_package:build_pip_package  --gpu` to have the `--project_name` of the `setup.py` of the package set to tensorflow_gpu:
See:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/pip_package/build_pip_package.sh#L57

**Comment by gunan** [comment]
Even if the pip packages are built with a different name, when installed they still occupy the same directory with the files with same names under your python library directories. Therefore, If you try to install both tensorflow and tensorflow-gpu you will overwrite the files the first installs with the second installation.
In the end, your installation may even not work.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by alkamid** [comment]
For anyone following @Mistobaan's advice (https://github.com/tensorflow/tensorflow/issues/11193#issuecomment-326819113), you should invoke `bazel build` normally, and only then use the `--gpu` arg:
```
bazel-bin/tensorflow/tools/pip_package/build_pip_package ${TF_ROOT}/pip/tensorflow_pkg --gpu
```
